devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,322 @@
+
+        /// <summary>
+        /// Magnetometer (R-HALL) temperature compensation value, used in axis compensation calculation functions
+        /// </summary>
+        private uint _rHall;
+
+        /// <summary>
+        /// Flag to evaluate disposal of resources
+        /// </summary>
+        private bool _shouldDispose = true;
+
+        /// <summary>
+        /// Gets or sets Magnetometer calibration compensation vector
+        /// </summary>
+        public Vector3 CalibrationCompensation { get; set; } = new Vector3();
+
+        /// <summary>
+        /// Primary I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L163
+        /// </summary>
+        public const byte PrimaryI2cAddress = 0x13;
+
+        /// <summary>
+        /// Secondary I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13, alhtough for m5stack is 0x10
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x10;
+
+        /// <summary>
+        /// Default timeout to use when timeout is not provided in the reading methods
+        /// </summary>
+        [Property]
+        public TimeSpan DefaultTimeout { get; set; } = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Default constructor for an independent Bmm150
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device</param>
+        public Bmm150(I2cDevice i2CDevice)
+            : this(i2CDevice, new Bmm150I2c())
+        {
+        }
+
+        /// <summary>
+        /// Constructor to use if Bmm150 is behind another element and need a special I2C protocol like
+        /// when used with the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="Bmm150Interface">The specific interface to communicate with the Bmm150</param>
+        /// <param name="shouldDispose">True to dispose the I2C device when class is disposed</param>
+        public Bmm150(I2cDevice i2cDevice, Bmm150I2cBase Bmm150Interface, bool shouldDispose = true)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _Bmm150Interface = Bmm150Interface;
+            _shouldDispose = shouldDispose;
+
+            Initialize();
+
+            // After initializing the device we read the 
+            _trimData = ReadTrimRegisters();
+        }
+
+        /// <summary>
+        /// Reads the trim registers of the sensor, used in compensation (x,y,z) calculation
+        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L1199
+        /// </summary>
+        /// <returns>Trim registers value</returns>
+        private Bmm150TrimRegisterData ReadTrimRegisters()
+        {
+            SpanByte trim_x1y1_data = new byte[2];
```

Comment:
missed those little ones with the naming convention. `trimX1Y1Data` but ok, that's not too bad anyway, your choice to fix it.